rustprocessor/src/handler/payload.rs

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2018 Intel corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------------
+ */
+
+use std::str;
+use std::fmt;
+
+use sawtooth_sdk::processor::handler::ApplyError;
+
+#[derive(Copy, Clone)]
+pub enum Action {
+    Deposit,
+    Withdraw,
+    Balance,
+    Transfer,
+}
+
+impl fmt::Display for Action {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "{}",
+            match *self {
+                Action::Deposit => "Action::Deposit",
+                Action::Withdraw => "Action::Withdraw",
+                Action::Balance => "Action::Balance",
+                Action::Transfer => "Action::Transfer",
+            }
+        )
+    }
+}
+
+
+pub struct SwPayload {
+    action: Action,    
+    value: u32,
+    beneficiary_pubkey: Option<String>,
+}
+
+impl SwPayload {
+
+    pub fn new(payload_data: &[u8]) -> Result<Option<SwPayload>, ApplyError> {
+    
+        let payload_string = match str::from_utf8(&payload_data) {
+            Ok(s) => s,
+            Err(_) => {
+                return Err(ApplyError::InvalidTransaction(String::from(
+                    "Invalid payload serialization",
+                )))
+            }
+        };
+
+        //SimpleWallet payload is constructed as comma separated items
+        let items: Vec<&str> = payload_string.split(",").collect();
+
+        if items.len() < 2 {
+            return Err(ApplyError::InvalidTransaction(String::from(
+                "Payload must have atleast 1 comma",
```

Comment:
You still need to fix this:
s/atleast/at least/

Reply:
ok